deps/Makefile

```diff
@@ -32,6 +32,11 @@ include $(SRCDIR)/llvm-ver.make
 
 ## Some shared configuration options ##
 
+# don't pass makefile flags through by default
+# (which would often result in unintentionally overriding the flags we are trying to set here)
+MAKEOVERRIDES =
```

Comment:
putting flags into the `CC` variable isn't terribly portable (aka `cmake` chokes on it). we should probably just remove that.